src/intervals/exact_numbers.jl

```diff
@@ -0,0 +1,68 @@
+struct ExactNumber{T} <: Real
+    number::T
+end
+
+# Promotion to Interval
+Base.promote_rule(::Type{Interval{T}}, ::Type{ExactNumber{S}}) where {T<:NumTypes,S<:Real} =
+    Interval{promote_numtype(T, S)}
+
+Base.promote_rule(::Type{ExactNumber{T}}, ::Type{Interval{S}}) where {T<:Real,S<:NumTypes} =
+    Interval{promote_numtype(T, S)}
+
+function Base.convert(::Type{Interval{T}}, x::ExactNumber) where {T<:NumTypes}
+    y = interval(T, x)
+    return _unsafe_interval(bareinterval(y), decoration(y), true)
+end
+
+# Promotion to Real
+Base.promote_rule(::Type{T}, ::Type{ExactNumber{S}}) where {T<:Real,S<:Real} =
+    promote_type(T, S)
+
+Base.promote_rule(::Type{ExactNumber{T}}, ::Type{S}) where {T<:Real,S<:Real} =
+    promote_type(T, S)
+
+Base.convert(::Type{T}, x::ExactNumber) where {T<:Real} =
+    convert(T, x.number)
```

Comment:
This is more a comment, or just a note. This would allow two `ExactNumber` to be converted, but this in principle can lead do not correct rounding. That being said, I am not opposed to this, we should just make it explicitly in the docstring

Reply:
Good point.

Because of the convert fallback normal reals can also silently be converted to `ExactReal`.

And that's not okay, so I will remove all conversions to `ExactReal`.